extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts

```diff
@@ -0,0 +1,338 @@
+		}).component());
+
+		this.podSummaryContainer = this.modelView.modelBuilder.divContainer().component();
+
+		this.refreshPodSummarySection();
+
+		content.addItem(this.podSummaryContainer);
+
+		// Pod Conditions
+		content.addItem(this.modelView.modelBuilder.text().withProps({
+			value: loc.podsPresent,
+			CSSStyles: { ...cssStyles.title }
+		}).component());
+
+		content.addItem(this.modelView.modelBuilder.text().withProps({
+			value: loc.podsUsedDescription,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px', 'margin-top': '10px' }
+		}).component());
+
+		this.podConditionsContainer = this.modelView.modelBuilder.divContainer().component();
+		this.podConditionsTable = this.modelView.modelBuilder.declarativeTable().withProps({
+			width: '100%',
+			columns: [
+				{
+					displayName: loc.condition,
+					valueType: azdata.DeclarativeDataType.string,
+					isReadOnly: true,
+					width: '20%',
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: cssStyles.tableRow
+				},
+				{
+					displayName: loc.details,
+					valueType: azdata.DeclarativeDataType.component,
+					isReadOnly: true,
+					width: '50%',
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: {
+						...cssStyles.tableRow,
+						'min-width': '150px'
+					}
+				},
+				{
+					displayName: loc.lastTransition,
+					valueType: azdata.DeclarativeDataType.string,
+					isReadOnly: true,
+					width: '30%',
+					headerCssStyles: cssStyles.tableHeader,
+					rowCssStyles: cssStyles.tableRow
+				}
+			],
+			data: [this.coordinatorData.map(p => [p.condition, p.details, p.lastUpdate])]
+		}).component();
+
+		this.podDropDown = this.modelView.modelBuilder.dropDown().withProps({ width: '150px' }).component();
+		this.disposables.push(
+			this.podDropDown.onValueChanged(() => {
+				this.podConditionsTable.setFilter(this.podConditionsTableIndexes.get(String(this.podDropDown.value)));
+			})
+		);
+
+		this.podConditionsContainer.addItem(this.podDropDown, { CSSStyles: { 'margin': '10px 0px 10px 0px' } });
+		this.podConditionsContainer.addItem(this.podConditionsTable);
+		this.podConditionsLoading = this.modelView.modelBuilder.loadingComponent()
+			.withItem(this.podConditionsContainer)
+			.withProperties<azdata.LoadingComponentProperties>({
+				loading: !this._postgresModel.configLastUpdated
+			}).component();
+
+		this.refreshPodCondtions();
+
+		content.addItem(this.podConditionsLoading, { CSSStyles: cssStyles.text });
+
+		this.initialized = true;
+		return root;
+	}
+
+	protected get toolbarContainer(): azdata.ToolbarContainer {
+		// Refresh
+		const refreshButton = this.modelView.modelBuilder.button().withProps({
+			label: loc.refresh,
+			iconPath: IconPathHelper.refresh
+		}).component();
+
+		this.disposables.push(
+			refreshButton.onDidClick(async () => {
+				refreshButton.enabled = false;
+				try {
+					this.podConditionsLoading!.loading = true;
+
+					await Promise.all([
```

Comment:
Since there's only 1 promise I think this can just be
`await this._postgresModel.refresh()`